integrated_channels/canvas/client.py

```diff
@@ -81,7 +83,24 @@ def _post(self, url, data):
         response = self.session.post(url, data=data)
         return response.status_code, response.text
 
-    def _delete(self, url, data):
+    def _put(self, data):
+        """
+        doc TBD
+        """
+        self._create_session()
+        integration_id = json.loads(
+            data.decode("utf-8")
+        )['course']['integration_id']
+        course_id = self._get_course_id_from_integration_id(integration_id)
```

Comment:
I think that in either of the following cases, we would like to raise a 400 Bad Request. A test can nicely cover these too

data is empty/None

data.decode fails and/or
  data.decode() does not contain 'course'->'integration_id'

course_id is None or cannot be extracted somehow